Source/WebCore/Modules/geolocation/Geolocation.cpp

```diff
@@ -345,18 +346,19 @@ static void logError(const String& target, const bool isSecure, const bool isMix
     
 bool Geolocation::shouldBlockGeolocationRequests()
 {
-    if (!isPermissionsPolicyAllowedByDocumentAndAllOwners(PermissionsPolicy::Feature::Geolocation, *document(), LogPermissionsPolicyFailure::Yes))
+    RefPtr document = this->document();
+    if (!isPermissionsPolicyAllowedByDocumentAndAllOwners(PermissionsPolicy::Feature::Geolocation, *document, LogPermissionsPolicyFailure::Yes))
         return true;
 
-    bool isSecure = SecurityOrigin::isSecure(document()->url()) || document()->isSecureContext();
-    bool hasMixedContent = !document()->foundMixedContent().isEmpty();
+    bool isSecure = SecurityOrigin::isSecure(document->url()) || document->isSecureContext();
+    bool hasMixedContent = !document->foundMixedContent().isEmpty();
     bool isLocalOrigin = securityOrigin()->isLocal();
-    if (document()->canAccessResource(ScriptExecutionContext::ResourceType::Geolocation) != ScriptExecutionContext::HasResourceAccess::No) {
+    if (document->canAccessResource(ScriptExecutionContext::ResourceType::Geolocation) != ScriptExecutionContext::HasResourceAccess::No) {
         if (isLocalOrigin || (isSecure && !hasMixedContent))
             return false;
     }
 
-    logError(securityOrigin()->toString(), isSecure, hasMixedContent, document());
+    logError(RefPtr { securityOrigin() }->toString(), isSecure, hasMixedContent, document.get());
```

Comment:
This will be an example of code that could benefit from having protectedSecurityOrigin() function.